### ISSUE TITLE
Annoying package-solution warning for extensions

### ISSUE THREAD

**Issue by wictorwilen** [SharePoint/sp-dev-docs#1350]
#### Category
- [ ] Question
- [ ] Typo
- [X] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
When packaging an SPFx solution using VSTS build pipelines I expect that the `gulp package-solution -- ship` command should work. and not abort after that task. The *warning* is nice, but should rather be textual information than a *warning*.

#### Observed Behavior
When having an application customizer in the solution the `package-solution` command gives the following warning:

> Warning - XXXApplicationCustomizer: Admins can make this solution available to all sites in the organization, but extensions won’t automatically appear. SharePoint Framework extensions must be specifically associated to sites, lists, and fields programmatically to be visible to site users.

This is interpreted by VSTS as an error and the build is aborted.

#### Steps to Reproduce

Create a simple VSTS build definition based on the `npm gulp`  template and add the `build`, `bundle` and `package-solution` npm tasks and see it fail on the `package-solution` task.

**Comment by nickpape-msft** [comment]
The idea of the warning was to let users know that they were in a potentially dangerous state, as their extensions will need to get explicitly enabled in the site collection level. However, this warning shouldn't be causing your VSTS builds to fail (it shouldn't be causing the build process to return a non-zero error code). Do you have the VSTS build definition configured to fail if something is written to stderr?

![image](https://user-images.githubusercontent.com/17131271/36129817-e7913210-101d-11e8-8759-4fb57398dbb2.png)

**Comment by wictorwilen** [comment]
Thanks @nickpape-msft. I understand why it's there - but it's more good thinking than real value. First of all I do not think that any admins will run this gulp command and thus never ever see it, so it kinda defeats its purpose.
For me this warning could be removed, and instead be shown at "install time". But if you want to keep it - just create a text message, no warning, no error.
To get this to work meanwhile, I have to use "Continue on Error" in VSTS (on the Gulp task). I can't se that "fail on standard error" you have. It works, but doesn't look pretty since the builds are only shown as partially succeeded.
![image](https://user-images.githubusercontent.com/5854279/36136704-11dd201e-1092-11e8-9bfc-a1df2cbd2b44.png)

**Comment by nickpape-msft** [comment]
One mitigation for now would be to add a warning suppression to the build.. e.g.

gulpfile.js
```js
build.addSuppression(/XXXApplicationCustomizer\: Admins can make this solution available to all sites in the organization/);
```

https://github.com/Microsoft/web-build-tools/blob/master/core-build/gulp-core-build/README.md#addsuppressionsuppression-string--regexp

**Comment by wictorwilen** [comment]
Tried that, but did not get that to work. Are you sure the suppressions works for the `package-solution` command?

**Comment by waldekmastykarz** [comment]
I've just tested it and it's working @wictorwilen. Here is my gulpfile.js

```js
'use strict';

const gulp = require('gulp');
const build = require('@microsoft/sp-build-web');
build.addSuppression(`Warning - [sass] The local CSS class 'ms-Grid' is not camelCase and will not be type-safe.`);
build.addSuppression(/Warning\s-\s[\w]+ApplicationCustomizer\: Admins can make this solution available to all sites in the organization/);

build.initialize(gulp);

```

**Comment by wictorwilen** [comment]
Gotta try it again. I did EXACTLY that the other day

**Comment by waldekmastykarz** [comment]
SPFx 1.4.0?

**Comment by wictorwilen** [comment]
It did work now, probably got that Regexp wrong then. And yes, it's 1.4. 

Thanks @waldekmastykarz and @nickpape-msft 

This should probably be added to the docs.

**Comment by nickpape-msft** [comment]
I've identified a longer-term fix for this issue which will make it so that the warnings that are written during this `package-solution` task do not cause the build to return a non-zero error code. We should see this in the next release.

**Comment by BryantAvey** [comment]
Thanks @nickpape-msft I was getting that warning all this week and was really confused by it.  Also I have a quick question, if anyone is willing to help out.  I built a simple Application Customizer to add a custom header. I've built and packaged it, and dropped the sppkg file up to the app catalog. I've published the app for all sites, but app is just not available to add from any site.  I've got other custom apps that do work, but this one just won't show up even though it shows enabled, valid, deployed and no errors the app catalog. 

What am I missing?
